### ISSUE TITLE
Cannot use a set of enums in the query builder

### ISSUE THREAD

**Issue by haikyuu** [edgedb/edgedb-js#289]
I'm getting the following error (after replacing `${a.__name__}` and `${b.__name__}` with their actual values manually
```
'Types are not castable: default::RoleEnum, default::RoleEnum'
```

The generated code seems to only allow sets of literals.

Schema
```
scalar type RoleEnum extending enum<Admin, Mod>
```

reproduction code `e.set(e.RoleEnum.Admin, e.RoleEnum.Mod)`

It seems like enums are not considered scalars in the query builder. I have patched it using this in `castMaps.ts`
```
if (a.__kind__ === "enum" && b.__kind__ === "enum") {
      if(b.__name__ === a.__name__) {
        return b;
      }
      throw new Error(`Types are not castable: ${a.__name__}, ${b.__name__}`);
      }
```

**Comment by haikyuu** [comment]
A similar error happens in the query builder for casting a set of enums 
```
  throw new Error(`Cannot convert literal '${type}' into scalar type`);
```

**Comment by colinhacks** [comment]
Fixed in https://github.com/edgedb/edgedb-js/pull/293 and landed in `edgedb@0.19.13`

**Comment by haikyuu** [comment]
working perfectly 🎉
